infrastructure/builder/src/generate/generators/docs-tocs.js

```diff
@@ -0,0 +1,204 @@
+const fs = require('fs');
+const md = require('md-directory');
+const readDirectory = require('read-directory');
+const { join } = require('path');
+const { readJSONSync, writeJSON } = require('../util');
+const removeExtension = require('../../../../../ui/src/utils/removeExtension');
+
+const DOCS_LOCATIONS = {
+  GENERATED: join('generated', 'docs'),
+  STATIC: join('ui', 'src', 'docs'),
+};
+
+const projectMetadata = {};
+function readProjectMetadata(name) {
+  if (!projectMetadata[name]) {
+    projectMetadata[name] = readJSONSync(join(DOCS_LOCATIONS.GENERATED, name, 'metadata.json'));
+  }
+
+  return projectMetadata[name];
+}
+
+// Sort doc files by the order property
+function sort(a, b) {
+  const first = a.data.order;
+  const second = b.data.order;
+
+  if (typeof first !== 'number') {
+    return 1;
+  }
+
+  if (typeof second !== 'number') {
+    return -1;
+  }
+
+  return first - second;
+}
+
+function sortChildren(children) {
+  // recursively sort child nodes
+  if (children && children.length) {
+    children.map(child => sortChildren(child.children));
+  }
+
+  children.sort(sort);
+}
+
+function readGeneratedDocs() {
+  return fs
+    .readdirSync(DOCS_LOCATIONS.GENERATED)
+    .reduce((acc, projectName) => {
+      const metadata = readProjectMetadata(projectName);
+      // We use md.parseDirSync instead of readDirectory in order
+      // to collect the front matter of the markdown file
+      const mdFiles = md.parseDirSync(join(DOCS_LOCATIONS.GENERATED, projectName), {
+        dirnames: true,
+      });
+      const jsonFiles = readDirectory.sync(
+        join(DOCS_LOCATIONS.GENERATED, projectName),
+        {
+          filter: 'references/*.json',
+          transform: content => ({
+            content: JSON.parse(content),
+            data: { inline: true },
+          }),
+        }
+      );
+      const files = { ...mdFiles, ...jsonFiles };
+
+      // Rename key
+      Object.keys(files).forEach(oldKey => {
+        const path = `reference/${metadata.tier}/${projectName}/${oldKey}`;
+
+        delete Object.assign(files, {
```

Comment:
🤔 how does this work? what is the intended effect?

Reply:
This rename the keys in order to include the section, tier, and service name that they belong to. This helps to provide a neat URL in the browser. I will add some comments :)